Frameworks/Foundation/NSInvocation.mm

```diff
@@ -268,3 +277,44 @@ - (void)invoke {
 }
 
 @end
+
+extern "C" void _NSInvocation_ForwardFrame(
+    void* stret, id self, SEL sel, void* frame, /* out */ _NSInvocationForwardReturnInfo* bridgeOut) {
+    NSMethodSignature* signature = [self methodSignatureForSelector:sel];
+    if (!signature) {
+        const char* types = nullptr;
+        types = sel_getType_np(sel);
+        if (!types) {
+            SEL typedSelector = nullptr;
+            sel_copyTypedSelectors_np(sel_getName(sel), &typedSelector, 1);
+            if (typedSelector) {
+                types = sel_getType_np(typedSelector);
+            }
+        }
+
+        if (types) {
+            signature = [NSMethodSignature signatureWithObjCTypes:types];
+        }
+    }
+
+    if (!signature) {
```

Comment:
I feel like it reads better this way :)

---
In reply to: [77266919](https://github.com/Microsoft/WinObjC/pull/951#discussion_r77266919) [](ancestors = 77266919)